### ISSUE TITLE
[Feature] Whitelist acceptable CSS styles

### ISSUE THREAD

**Issue by mrnoname1000** [FreshRSS/FreshRSS#4738]
**Is your feature request related to a problem? Please describe.**
FreshRSS currently strips all CSS styles from incoming HTML. However, some attributes (e.g. color, font-weight, border) contribute to the structure of the content. W3C's validator service has [some recommendations](https://validator.w3.org/feed/docs/warning/DangerousStyleAttr.html) on which tags are safe to allow.

**Describe the solution you’d like**
FreshRSS filters unsafe attributes from `style=""` HTML attributes, leaving the ones that contribute to the content. Additionally, for `<style>` tags, scope the selectors so they only apply to a single article.

**Describe alternatives you’ve considered**
Some CSS attributes could be replaced with equivalent HTML tags (e.g. `<font>`). For an HTML5 frontend there's not much reason to do this, but maybe this would be preferable for API clients in case they do their own sanitizing.

**Additional context**
40 of the 80 sites from which I follow feeds use some sort of inline style. Some aggregate user-generated content from a WYSIWYG editor (e.g. [MyAnimeList](https://myanimelist.net/rss.php?type=blog)), some use `<style>` tags (e.g. [TorrentFreak](https://torrentfreak.com/feed/), [DuckDuckGo](https://spreadprivacy.com/rss/), [OpenAI](https://openai.com/blog/rss/)), and many others use `style=""` HTML attributes (e.g. [GitHub Releases](https://github.com/FreshRSS/FreshRSS/releases.atom), [Invidious](https://invidious.snopyta.org/feed/channel/UCBR8-60-B28hp2BmDPdntcQ), too many others to list here). I haven't found a feed yet that uses linked stylesheets.

**Comment by aledeg** [comment]
Could you extract some feed examples and attach them in the current issue?
That will help us work on that even if the feed you are referring to are not working anymore.
Thank you.

See https://github.com/FreshRSS/FreshRSS/blob/edge/docs/en/developers/06_Reporting_Bugs.md#how-to-provide-feed-data

**Comment by mrnoname1000** [comment]
Sure thing:
- `<style>`:
  - [torrentfreak](https://github.com/FreshRSS/FreshRSS/files/9803999/torrentfreak.rss.xml.txt)
  - [duckduckgo](https://github.com/FreshRSS/FreshRSS/files/9804009/duckduckgo.rss.xml.txt)
  - [openai](https://github.com/FreshRSS/FreshRSS/files/9804016/openai.rss.xml.txt)
- `style=""`:
  - [github](https://github.com/FreshRSS/FreshRSS/files/9804885/github.atom.xml.txt)
  - [invidious](https://github.com/FreshRSS/FreshRSS/files/9804031/invidious.atom.xml.txt)
- WYSIWYG (`style=""`)
  - [myanimelist](https://github.com/FreshRSS/FreshRSS/files/9803969/myanimelist.rss.xml.txt)

**Comment by Alkarex** [comment]
Hum, that would be quite challenging to support.
I cannot see the example of use in a GitHub release.

**Comment by mrnoname1000** [comment]
Yeah, it would probably require a CSS parser/serializer or some intelligent filtering. [Sabberworm's parser](https://github.com/sabberworm/PHP-CSS-Parser) looks like a well maintained but fairly large dependency.

Looks like the only style in that GitHub feed is used for max-width of an image. Here are a couple other feeds that use them more extensively:
[blog.ajay.app](https://blog.ajay.app/rss.xml) ([attachment](https://github.com/FreshRSS/FreshRSS/files/9804948/blog.ajay.app.rss.xml.txt))
[NASA blogs](https://blogs.nasa.gov/feed) ([attachment](https://github.com/FreshRSS/FreshRSS/files/9804961/nasa.rss.xml.txt))
[Jellyfin](https://jellyfin.org/index.xml) ([attachment](https://github.com/FreshRSS/FreshRSS/files/9804964/jellyfin.rss.xml.txt))

Also Invidious feeds have broken text in FreshRSS because they use `white-space: pre-wrap` instead of a `<pre>` tag, but I would argue that's an issue on their end.

**Comment by math-GH** [comment]
Does the CSP block inline styles?

**Comment by Alkarex** [comment]
> Does the CSP block inline styles?

Yes, our CSP rules block inline styles at the moment for security reasons

**Comment by math-GH** [comment]
> > Does the CSP block inline styles?
> 
> Yes, our CSP rules block inline styles at the moment for security reasons

I would keep it. And it means that there is no chance to contribute the styles, isn't it?

**Comment by Alkarex** [comment]
Even without CSP, that would be pretty challenging. As @Frenzie writes, it would require some kind of parsing, and CSS can be abused in many ways. Or we would need some kind of container such as an `<iframe>` or similar.

**Comment by Frenzie** [comment]
Hm, what did I write? :-)

**Comment by Alkarex** [comment]
> Hm, what did I write? :-)

Haha, sorry, it was @mrnoname1000
